engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/encode/OutputValidator.scala

```diff
@@ -0,0 +1,32 @@
+package pl.touk.nussknacker.engine.avro.encode
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import org.apache.avro.Schema
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CustomNodeError, NodeId}
+import pl.touk.nussknacker.engine.api.typed.{CanBeSubclassDeterminer, SingleSubclassRestriction, typing}
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedObjectTypingResult, TypingResult}
+import pl.touk.nussknacker.engine.avro.KafkaAvroBaseTransformer.SinkValueParamName
+import pl.touk.nussknacker.engine.avro.typed.AvroSchemaTypeDefinitionExtractor
+
+object OutputValidator {
+
+  def validateOutput(value: TypingResult, schema: Schema, validationMode: ValidationMode)(implicit nodeId: NodeId): Validated[CustomNodeError, Unit] = {
+    val possibleTypes = AvroSchemaTypeDefinitionExtractor.ExtendedPossibleTypes
+    //FIXME: this still does not handle optional fields validation properly for acceptUnfilledOptional == true?
+    val returnType = new AvroSchemaTypeDefinitionExtractor(skippNullableFields = validationMode.acceptUnfilledOptional).typeDefinition(schema, possibleTypes)
```

Comment:
I know that it is not part of your change, but `skipNullableFields` is wrong name because we skip `optional` fields. By optional we should understand field with default value. It can be filed without union with null but with default value.

Reply:
ok